### DISCUSSION TITLE
Trying to load messages per page on getRequestConfig.

### DISCUSSION THREAD

**Discussion by TivonJJ** [amannn/next-intl#1088] [Q&A]
My site have a big internationalized messages because there are many pages, I see the getRequestConfig is load full messages in the demo.  So i got a big data when page initialization,this affects the speed of response. I tried to load by the current page.

```javascript
// i18n.ts
export default getRequestConfig(async ({ locale }) => {
    if (!config.locales.includes(locale)) notFound();
    // eg.  /about
    const page = headers().get('x-middleware-pathname');
    const messages = loadMessages(locale, page);
    return {
        messages,
    };
});
```

```javascript
// layout.tsx
    const messages = await getMessages({ locale });
    ....
    return <NextIntlClientProvider locale={locale} messages={messages}>
.....
</NextIntlClientProvider>
```

It works fine when the page first loads,  But when redirect the page, Or use your browser's forward and backward,The text is lost.

I debugged it and found the problem： When redirect/forward/backward  The layout.tsx was not executed, so the getRequestConfig was not invoked. there intl message is not newest,this resulted in the loss of messages on my page.

How should I solve this problem? Or is there a better solution for loading messages on a per-page basis?

**Comment by amannn** [comment]
Currently it's expected that you return all messages from `i18n.ts`, but you can be picky about the ones you pass to the client side via `NextIntlClientProvider` (please see the docs).
